pkg/controllers/management/restrictedadminrbac/fleet_test.go

```diff
@@ -0,0 +1,277 @@
+								OwnerReferences: ownerRefs,
+							},
+							RoleRef:  roleRef,
+							Subjects: subjects,
+						}, nil
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					UpdateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						assert.Equal(t, rb, expected)
+						return expected, nil
+					},
+				}
+			},
+			wantErr: false,
+		},
+		{
+			name: "one previously existing incorrect rolebinding (wrong OwnerReferences)",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return &k8srbac.RoleBinding{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:            name,
+								Namespace:       namespace,
+								Labels:          map[string]string{rbac.RestrictedAdminClusterRoleBinding: "true"},
+								OwnerReferences: []metav1.OwnerReference{},
+							},
+							RoleRef:  roleRef,
+							Subjects: subjects,
+						}, nil
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					UpdateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						assert.Equal(t, rb, expected)
+						return expected, nil
+					},
+				}
+			},
+			wantErr: false,
+		},
+		{
+			name: "one previously existing incorrect rolebinding (wrong RoleRef)",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return &k8srbac.RoleBinding{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:            name,
+								Namespace:       namespace,
+								Labels:          map[string]string{rbac.RestrictedAdminClusterRoleBinding: "true"},
+								OwnerReferences: ownerRefs,
+							},
+							RoleRef:  k8srbac.RoleRef{},
+							Subjects: subjects,
+						}, nil
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					UpdateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						assert.Equal(t, rb, expected)
+						return expected, nil
+					},
+				}
+			},
+			wantErr: false,
+		},
+		{
+			name: "one previously existing incorrect rolebinding (wrong Subjects)",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return &k8srbac.RoleBinding{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:            name,
+								Namespace:       namespace,
+								Labels:          map[string]string{rbac.RestrictedAdminClusterRoleBinding: "true"},
+								OwnerReferences: ownerRefs,
+							},
+							RoleRef:  roleRef,
+							Subjects: []k8srbac.Subject{},
+						}, nil
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					UpdateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						assert.Equal(t, rb, expected)
```

Comment:
Here, and in the other places in the file, you are using `t` as the `*testing.T` arg. This is the test arg for the entire test function, so if you have a failure here it can only tell you that the entire test failed. For example, if I change this line to `assert.NotEqual(t, rb, expected)`, I see:

```
--- FAIL: Test_rbaccontroller_ensureRolebinding (0.00s)
    fleet_test.go:190: 
...
                Test:           Test_rbaccontroller_ensureRolebinding

```
This makes it hard to figure out exactly what sub-test failed. However, if you change `t` to `c.t` (which is the `testing.T` object for this sub-test, set on the mockController when you constructed that), you see:

```
--- FAIL: Test_rbaccontroller_ensureRolebinding (0.00s)
    --- FAIL: Test_rbaccontroller_ensureRolebinding/one_previously_existing_incorrect_rolebinding_(wrong_Subjects) (0.00s)
        fleet_test.go:190:
...
                Test:           Test_rbaccontroller_ensureRolebinding/one_previously_existing_incorrect_rolebinding_(wrong_Subjects)
```

Which tells you exactly what sub-test failed, making it much easier to track down failures.

In short:
- Change `t` to `c.t`, both here and in other places where this used.

Reply:
Thanks for the mini-lesson, really appreciated the detail and clarity. Addressed as 83b064d2b